static/app/components/searchQueryBuilder/tokens/filter/valueCombobox.tsx

```diff
@@ -377,10 +378,26 @@ function useFilterSuggestions({
   );
 
   const suggestionGroups: SuggestionSection[] = useMemo(() => {
-    return shouldFetchValues
-      ? [{sectionText: '', suggestions: data?.map(value => ({value})) ?? []}]
-      : (predefinedValues ?? []);
-  }, [data, predefinedValues, shouldFetchValues]);
+    if (!shouldFetchValues) {
+      return predefinedValues ?? [];
+    }
+
+    const suggestions = data?.map(value => {
+      return {
+        value,
+        description:
+          // When the key is device, we can help users by displaying the readable name
+          key?.key === 'device' ? (
```

Comment:
Hmm I'm surprised we don't have a `FieldKey.DEVICE`. Can you add one in `utils/fields/index.tsx`? We seem to have enums for device.class and others